internal/action/config.go

```diff
@@ -80,20 +81,42 @@ func contains(haystack []string, needle string) bool {
 func (s *Action) setConfigValue(ctx context.Context, store, key, value string) error {
 	debug.Log("setting %s to %s for %q", key, value, store)
 
-	if err := s.cfg.Set(store, key, value); err != nil {
+	level, err := s.cfg.SetWithLevel(store, key, value)
+	if err != nil {
 		return fmt.Errorf("failed to set config value %q: %w", key, err)
 	}
 
+	// in case of a non-local config change we don't need to track changes in the store
+	if level != config.Local {
+		debug.Log("did not set local config (was config level %d) in store '%s', skipping commit phase", level, store)
+		s.printConfigValues(ctx, store, key)
+
+		return nil
+	}
+
 	st := s.Store.Storage(ctx, store)
 	if st == nil {
 		return fmt.Errorf("storage not available")
 	}
 
-	if err := st.Add(ctx, "config"); err != nil && !errors.Is(err, istore.ErrGitNotInit) {
-		return fmt.Errorf("failed to stage config file: %w", err)
+	// notice that we rely on the cfg.Set above having created the local config file here. If it doesn't exist,
```

Comment:
Great comment. But one idea: Should we maybe check if the file exists and gracefully return if not?

Reply:
That's what I was doing earlier, happy to add it back.